### PR TITLE
DR-2574 Template for homepage

### PR DESCRIPTION
**Pull request by 7emansell**
## Ticket:

Fixes JIRA ticket [DR-2574](https://jira.nypl.org/browse/DR-2574)

## This PR does the following:

- Adds the template to the DC homepage

## How has this been tested?

Tested locally.

## Accessibility concerns or updates

N/A.

### Checklist:

- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### CODE REVIEW

File: __tests__/homepage.test.tsx
Comment (7emansell) [inline __tests__/homepage.test.tsx:1]: Best I could come up with for unit tests on components that don't actually have any content, but let me know if there's a more thorough way to do this

File: __tests__/homepage.test.tsx [hunk 1]
```diff
@@ -17,4 +17,15 @@ describe("Home", () => {
     });
     expect(header).toBeInTheDocument();
   });
+
+  it("renders the expected text content", () => {
+    const { getByText } = render(<Home />);
```
Comment (7emansell) [inline __tests__/homepage.test.tsx:22]: Best I could come up with for unit tests on components that don't actually have any content, but is there a more thorough way to do this?

Comment (EdwinGuzman) [reply __tests__/homepage.test.tsx:22]: This is okay! We also had a hard time testing this on the DS. The accessibility test covers what we want as well.

File: src/pages/index.tsx [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+  TemplateContent,
+  TemplateFooter,
+  TemplateHeader,
+} from "@nypl/design-system-react-components";
 
 export default function Home() {
-  return <CampaignHero />;
+  return (
+    <>
+      <SkipNavigation />
```
Comment (EdwinGuzman) [inline src/pages/index.tsx:15]: Thanks for adding this, I forgot about this. It also needs a bit more clarity from Clare or Lana about where the "Skip to Main Content" link should go to. The link goes to `#mainContent` but there's no element on the page with this id. Can you add it to the h1 in the Hero (currently has `campaign-hero`) for now?

We can bring this up once the swim lanes are complete -- should we point to the hero heading or the first swim lane?

File: src/pages/index.tsx [hunk 3]
```diff
@@ -1,5 +1,38 @@
...
+      <Template>
```
Comment (EdwinGuzman) [inline src/pages/index.tsx:16]: Just curious, did you try the `TemplateAppContainer` component as well? You don't need to change it but just wondering if there were any issues with it.

File: src/pages/index.tsx [hunk 4]
```diff
@@ -1,5 +1,38 @@
...
+          </TemplateHeader>
+          <CampaignHero />
+        </TemplateBreakout>
+        <TemplateContent>
+          <p>First swim lane</p>
+          <p>Featured Content</p>
+          <p>Rest of swim lanes</p>
+          <p>Explore further links</p>
+        </TemplateContent>
+        <TemplateFooter>
```
Comment (EdwinGuzman) [inline src/pages/index.tsx:32]: The NYPL Footer will go in a different location so this is okay to remove.

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Just had one question about the `TemplateAppContainer` component and two minor updates.

File: src/pages/index.tsx (hunk 4 above)
Comment (7emansell) [reply src/pages/index.tsx]: Does this mean remove the template footer entirely?

File: src/pages/index.tsx (hunk 3 above)
Comment (7emansell) [reply src/pages/index.tsx]: Refactored to use `TemplateAppContainer`

File: src/pages/index.tsx (hunk 4 above)
Comment (EdwinGuzman) [reply src/pages/index.tsx]: Yes, correct.

**Review by EdwinGuzman — APPROVED** [review]

**Review by avertrees — APPROVED** [review]

**Pull request merged**